pkg/antctl/raw/multicluster/init.go

```diff
@@ -0,0 +1,169 @@
+	clusterID  string
+	secret     string
+	output     string
+}
+
+var initOpts *initOptions
+
+func (o *initOptions) validate() error {
+	if o.namespace == "" {
+		return fmt.Errorf("Namespace is required")
+	}
+	if o.clusterSet == "" {
+		return fmt.Errorf("ClusterSet is required")
+	}
+	if o.clusterID == "" {
+		return fmt.Errorf("ClusterID is required")
+	}
+	return nil
+}
+
+var initExample = strings.Trim(`
+# Initialize ClusterSet in the given Namespace of the leader cluster.
+  $ antctl mc init --namespace antrea-multicluster --clusterset clusterset --clusterid cluster-north
+# Initialize ClusterSet in the given Namespace of the leader cluster and output the config file for the member cluster.
+  $ antctl mc init --namespace antrea-multicluster --clusterset clusterset --clusterid cluster-north --output-file cluster-config.yaml
+# Initialize ClusterSet in the given Namespace of the leader cluster and create a secret for the member cluster with a Secret.
+  $ antctl mc init --namespace antrea-multicluster --clusterset clusterset --clusterid cluster-north --secret cluster-secret
+`, "\n")
+
+func NewInitCommand() *cobra.Command {
+	command := &cobra.Command{
+		Use:     "init",
+		Short:   "Initialize ClusterSet in the given Namespace of the leader cluster",
+		Args:    cobra.MaximumNArgs(0),
+		Example: initExample,
+		RunE:    initRunE,
+	}
+
+	o := initOptions{}
+	initOpts = &o
+	command.Flags().StringVarP(&o.namespace, "namespace", "n", "", "Antrea Multi-cluster Namespace")
+	command.Flags().StringVarP(&o.clusterSet, "clusterset", "", "", "ClusterSet ID of the leader cluster")
+	command.Flags().StringVarP(&o.clusterID, "clusterid", "", "", "ClusterID of the leader cluster")
+	command.Flags().StringVarP(&o.secret, "secret", "", "", "If specified, a Secret will be created for joining the member cluster")
+	command.Flags().StringVarP(&o.output, "output-file", "o", "", "Output file to save the member cluster join config")
+
+	return command
+}
+
+func initRunE(cmd *cobra.Command, args []string) error {
+	if err := initOpts.validate(); err != nil {
+		return err
+	}
+	k8sClient, err := common.NewClient(cmd)
+	if err != nil {
+		return err
+	}
+	createdRes := []map[string]interface{}{}
+	var createErr error
+	defer func() {
+		if createErr != nil {
+			fmt.Fprintf(cmd.OutOrStderr(), "Failed to init the Antrea Multi-cluster. Deleting the created resources\n")
+			if err := common.Rollback(cmd, k8sClient, createdRes); err != nil {
+				fmt.Fprintf(cmd.OutOrStdout(), "Failed to rollback: %v\n", err)
+			}
+		}
+	}()
+	createErr = common.CreateClusterClaim(cmd, k8sClient, initOpts.namespace, initOpts.clusterSet, initOpts.clusterID, &createdRes)
+	if createErr != nil {
+		return createErr
+	}
+	createErr = common.CreateClusterSet(cmd, k8sClient, initOpts.namespace, initOpts.clusterSet, "", "", "", initOpts.clusterID, initOpts.namespace, &createdRes)
+	if createErr != nil {
+		return createErr
+	}
+
+	var file *os.File
+	if initOpts.output != "" {
```

Comment:
Should we output the file to a temporary path (maybe current working directory) if user didn't set this option?

Reply:
I'm not sure if we need a default output dir. The users may not use the config file but use the command to join the leader cluster.